src/pages/News/container/newsContainer.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './news.css';
+
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
+import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos';
+import LanguageIcon from '@mui/icons-material/Language';
+import { Button, Grid, Typography, Dialog, DialogTitle, DialogContent, DialogActions } from '@mui/material';
+import Footer from '../../Footer/footer';
+import ClearIcon from '@mui/icons-material/Clear';
+
+const newsContainer = () => {
+    const [itemActive, setItemActive] = useState(0);
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [dialogData, setDialogData] = useState(null);
+
+  const thumbnailRef = useRef(null);
+
+  const nextClick = () => {
+    setItemActive((prevItemActive) => (prevItemActive + 1) % items.length);
+  };
+
+  const prevClick = () => {
+    setItemActive((prevItemActive) => (prevItemActive - 1 + items.length) % items.length);
+  };
+
+  const thumbnailClick = (index) => {
+    setIsTimerPaused(true); // Pause the timer when a click is registered
+    setItemActive(index);
+    setDialogOpen(true);
+    setDialogData(items[index]);
+  };
+
+  const handleClose = () => {
+    setIsTimerPaused(false); // Resume the timer when the dialog is closed
+    setDialogOpen(false);
+  };
+
+  const [isTimerPaused, setIsTimerPaused] = useState(false);
+
+  useEffect(() => {
+    const refreshInterval = setInterval(() => {
+      if (!isTimerPaused) {
+        nextClick();
+      }
+    }, 3000);
+    return () => clearInterval(refreshInterval);
+  }, [isTimerPaused]);
+
+  useEffect(() => {
+    if (thumbnailRef.current) {
```

Comment:
what is this used for? can you give some comments?

Reply:
I've used the thumnailRef to design my own carousel slider which was deigned .

Reply:
the design of that cards floating 
[
<img width="642" alt="1" src="https://github.com/varunnivas/newstreet-website-proj/assets/162587587/f807c1b3-1921-4fae-bbeb-8022cd40aab1">
](url)